include/cpp2util.h

```diff
@@ -1629,39 +1629,44 @@ constexpr auto is( X&& ) {
 //  Types
 //
 template< typename C, typename X >
-auto is( X const& x ) -> bool {
+constexpr auto is( X const& x ) -> auto {
     if constexpr (
         std::is_same_v<C, X>
         || std::is_base_of_v<C, X>
     )
     {
-        return true;
+        return std::true_type{};
```

Comment:
What are the benefits of returning `std::true_type` instead of `bool`?

My familiarity with `std::true_type` and `std::false_type` is with TMP, and in general the compile-time work we're doing here tries to avoid TMP.

Reply:
I commented on the same question below.

Reply:
Answered here: https://github.com/hsutter/cppfront/pull/1204#discussion_r1712342613